profiling/src/profile/mod.rs

```diff
@@ -58,6 +57,38 @@ impl From<&SampleId> for u64 {
 #[repr(transparent)]
 pub struct StackTraceId(usize);
 
+#[derive(Copy, Clone, Default, Debug, Eq, PartialEq, Hash)]
+#[repr(transparent)]
+pub struct LocationId(u32);
```

Comment:
Thoughts on using `NonZeroU32` instead of `u32` since we literally are specifically avoiding 0 anyway?

Reply:
Good idea.  Will make `Option<LocationId>` cheaper